### PR TITLE
feat: add Tau simulated user strategy

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary
- implement Tau strategy using SimulatedUser provider
- expose redteam provider `promptfoo:redteam:tau`
- document Tau in strategy docs and presets
- support Tau in redteam setup UI
- add unit tests

## Testing
- `npx prettier -w ...`
- `npx eslint ...`
- `npm test` *(fails: AwsBedrock tests due to missing deps)*

### CODE REVIEW

**Review by typpo — APPROVED** [review]

**Review by typpo — COMMENTED** [review]
Can we just call it simulated-user and not tau?  like the plugin

**Comment by sklein12** [comment]
when selecting this strategy the "Stateful configuration" section needs to be displayed 
image
image

**Comment by sklein12** [comment]
This is failing when `stateful` is set to true for a target. It's sending the full conversation history instead of just the latest message. You can use the example target to reproduce.

**Review by sklein12 — CHANGES_REQUESTED** [review]
1. Needs to show the stateful configuration UI
2. Needs to support stateful true/false
3. Needs to support sessionIds

File: src/providers/simulatedUser.ts [hunk 1]
```diff
@@ -60,14 +79,32 @@ export class SimulatedUser implements ApiProvider {
     targetProvider: ApiProvider,
     prompt: string,
   ): Promise<Message[]> {
+    logger.debug('[SimulatedUser] Sending message to target provider');
+
+    const payload = this.targetConfig.stateful
+      ? JSON.stringify([{ role: 'system', content: prompt }])
```
Comment (sklein12) [inline src/providers/simulatedUser.ts]: does goat and crescendo just said the prompt or send it in OpenAI format like this?

File: src/providers/simulatedUser.ts [hunk 2]
```diff
@@ -29,10 +30,26 @@ export class SimulatedUser implements ApiProvider {
...
+  };
+
   constructor({ id, label, config }: AgentProviderOptions) {
     this.identifier = id ?? label ?? 'agent-provider';
     this.maxTurns = config.maxTurns ?? 10;
     this.rawInstructions = config.instructions || '{{instructions}}';
+
+    this.targetConfig = {
+      stateful: config.stateful ?? false,
+      sessionId: null,
```
Comment (sklein12) [inline src/providers/simulatedUser.ts]: SessionId is not a config value, it's passed in vars. Please look at how its implemented in Goat and Crescendo and follow the same pattern.

Comment (will-holley) [reply src/providers/simulatedUser.ts]: That's how it's used.

The other providers pass the `context` into the caller scope – which would be `sendMessageToAgent` here – and scope it onto `context.vars`. I tried to minimize changes to the call signatures and thus scoped it instead / placed it under `targetConfig` because it describes the configuration of the target i.e. the target is configured with a session.

File: src/providers/simulatedUser.ts (hunk 1 above)
Comment (will-holley) [reply src/providers/simulatedUser.ts]: They don't send it in the OpenAI format.

File: src/providers/simulatedUser.ts (hunk 2 above)
Comment (sklein12) [reply src/providers/simulatedUser.ts]: Please follow the same pattern as the others. It is a var not a config option.

File: src/providers/simulatedUser.ts (hunk 1 above)
Comment (sklein12) [reply src/providers/simulatedUser.ts]: Why are you doing it differently here?

**Comment by will-holley** [comment]
Related: https://github.com/promptfoo/promptfoo-cloud/pull/1511

File: src/providers/simulatedUser.ts [hunk 3]
```diff
@@ -21,28 +22,44 @@ export type AgentProviderOptions = ProviderOptions & {
...
   id() {
     return this.identifier;
   }
 
+  /**
+   * Non-Redteam SimulatedUserProvider uses the 'tau' task.
+   */
+  get taskId() {
+    return 'tau';
+  }
```
Comment (sklein12) [inline src/providers/simulatedUser.ts]: what's this used for? I can't find it anywhere in the codebase

File: src/providers/simulatedUser.ts [hunk 4]
```diff
@@ -21,28 +22,44 @@ export type AgentProviderOptions = ProviderOptions & {
...
+
   private async sendMessageToUser(
     messages: Message[],
     userProvider: PromptfooSimulatedUserProvider,
   ): Promise<Message[]> {
+    logger.debug('[SimulatedUser] Sending message to simulated user (tau) provider');
```
Comment (sklein12) [inline src/providers/simulatedUser.ts]: What's the purpose of this message? Did you mean to include the message?

File: src/redteam/constants/metadata.ts [hunk 5]
```diff
@@ -122,6 +123,7 @@ export const subCategoryDescriptions: Record<Plugin | Strategy, string> = {
   rbac: 'Tests role-based access control implementation',
   'reasoning-dos':
     'Tests for vulnerabilities to computational resource exhaustion through excessive reasoning patterns',
+  [REDTEAM_SIMULATED_USER_STRATEGY_ID]: 'Simulated user conversations over multiple turns',
```
Comment (sklein12) [inline src/redteam/constants/metadata.ts]: Could you add some more description here? In reality everything we do is a simulated user.  What's a more precise way to say what this strategy is?

File: src/providers/simulatedUser.ts (hunk 4 above)
Comment (will-holley) [reply src/providers/simulatedUser.ts]: Yeah, to improve clarity in the logs

File: src/providers/simulatedUser.ts (hunk 3 above)
Comment (will-holley) [reply src/providers/simulatedUser.ts]: `SimulatedUser` and `RedteamSimulatedUserProvider` call different server tasks

File: site/docs/_shared/data/strategies.ts [hunk 6]
```diff
@@ -135,6 +135,16 @@ export const strategies: Strategy[] = [
     asrIncrease: '70-90%',
     link: '/docs/red-team/strategies/goat/',
   },
+  {
+    category: 'Multi-turn',
+    strategy: 'mischievous-user',
+    displayName: 'Mischievous User',
+    description: 'Mischievous user conversations',
+    longDescription: 'Simulates a multi-turn conversation between a mischievous user and an agent',
+    cost: 'Low',
```
Comment (mldangelo) [inline site/docs/_shared/data/strategies.ts]: high

File: src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx [hunk 7]
```diff
@@ -85,7 +93,8 @@ export default function StrategyConfigDialog({
       strategy === 'crescendo' ||
       strategy === 'pandamonium' ||
       strategy === 'gcg' ||
-      strategy === 'citation'
+      strategy === 'citation' ||
+      strategy === REDTEAM_MISCHIEVOUS_USER_STRATEGY_ID
```
Comment (mldangelo) [inline src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx]: I feel like we should just use the string. I find this more confusing.

File: test/redteam/strategies/strategyId.test.ts [hunk 8]
```diff
@@ -94,6 +94,7 @@ describe('Strategy IDs', () => {
       'prompt-injection': 'promptInjections/index.ts',
       retry: 'retry.ts',
       rot13: 'rot13.ts',
+      'mischievous-user': 'mischievousUser.ts',
```
Comment (mldangelo) [inline test/redteam/strategies/strategyId.test.ts]: please keep the alphabetical order

File: test/providers/simulatedUser.test.ts [hunk 9]
```diff
@@ -78,14 +78,7 @@ describe('SimulatedUser', () => {
       expect(result.output).toContain('User:');
       expect(result.output).toContain('Assistant:');
       expect(result.tokenUsage?.numRequests).toBe(2);
-      expect(originalProvider.callApi).toHaveBeenCalledWith(
```
Comment (mldangelo) [inline test/providers/simulatedUser.test.ts:81]: prefer not to modify this file if the tests still pass

File: test/providers/registry.test.ts [hunk 10]
```diff
@@ -162,6 +162,7 @@ describe('Provider Registry', () => {
         'promptfoo:redteam:best-of-n',
         'promptfoo:redteam:crescendo',
         'promptfoo:redteam:goat',
+        'promptfoo:redteam:mischievous-user',
```
Comment (mldangelo) [inline test/providers/registry.test.ts]: please sort

File: src/redteam/strategies/index.ts [hunk 11]
```diff
@@ -110,6 +112,15 @@ export const Strategies: Strategy[] = [
       return newTestCases;
     },
   },
+  {
+    id: REDTEAM_MISCHIEVOUS_USER_STRATEGY_ID,
+    action: async (testCases, injectVar, config) => {
+      logger.debug(`Adding simulated user test cases to ${testCases.length} test cases`);
```
Comment (mldangelo) [inline src/redteam/strategies/index.ts]: update the logs

File: src/redteam/providers/mischievousUser.ts [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        stateful: config.stateful ?? false,
+      },
+    });
+  }
+
+  id() {
+    return PROVIDER_ID;
+  }
+
+  get taskId() {
```
Comment (mldangelo) [inline src/redteam/providers/mischievousUser.ts]: What is taskId? Do we need it / can we get rid of it?

File: src/providers/simulatedUser.ts [hunk 13]
```diff
@@ -100,32 +140,50 @@ export class SimulatedUser implements ApiProvider {
...
         context.originalProvider,
-        prompt,
         context,
       );
-      messages = messagesToAgent;
-      numRequests += 1; // Only count the request to the agent.
+
+      messages.push({ role: 'assistant', content: String(agentResponse.output ?? '') });
+
+      if (agentResponse.tokenUsage) {
```
Comment (mldangelo) [inline src/providers/simulatedUser.ts:153]: Unfortunately this isn't complete but it's not complete in a lot of places. I think good enough for now

File: src/providers/promptfoo.ts [hunk 14]
```diff
@@ -192,9 +192,11 @@ interface PromptfooAgentOptions {
 
 export class PromptfooSimulatedUserProvider implements ApiProvider {
   private options: PromptfooAgentOptions;
+  private taskId: string;
 
-  constructor(options: PromptfooAgentOptions = {}) {
+  constructor(options: PromptfooAgentOptions = {}, taskId: string) {
     this.options = options;
+    this.taskId = taskId;
```
Comment (mldangelo) [inline src/providers/promptfoo.ts:199]: preference would be to remove this.

**Review by mldangelo — COMMENTED** [review]
Thank you for working on this! It's very close to ready

File: src/redteam/providers/mischievousUser.ts (hunk 12 above)
Comment (will-holley) [reply src/redteam/providers/mischievousUser.ts]: This class inherits from `SimulatedUser` and uses a different redteam task.  These need to be passed into the `PromptfooSimulatedUserProvider` server call for their respective `callApi` requests to be correctly routed.

File: src/providers/promptfoo.ts (hunk 14 above)
Comment (will-holley) [reply src/providers/promptfoo.ts:199]: See https://github.com/promptfoo/promptfoo/pull/4107#discussion_r2190952787

File: test/providers/simulatedUser.test.ts (hunk 9 above)
Comment (will-holley) [reply test/providers/simulatedUser.test.ts:81]: They don't

File: src/providers/simulatedUser.ts (hunk 4 above)
Comment (sklein12) [reply src/providers/simulatedUser.ts]: How so?

File: test/providers/simulatedUser.test.ts (hunk 9 above)
Comment (mldangelo) [reply test/providers/simulatedUser.test.ts:81]: Does the behavior of simulated user change / did you test it?

File: test/redteam/providers/simulatedUser.test.ts [hunk 15]
```diff
@@ -0,0 +1,228 @@
+import { SimulatedUser } from '../../../src/providers/simulatedUser';
+import RedteamSimulatedUserProvider from '../../../src/redteam/providers/mischievousUser';
+import type { CallApiContextParams, CallApiOptionsParams, Prompt } from '../../../src/types';
+
+jest.mock('../../../src/providers/simulatedUser');
+
+describe('RedteamSimulatedUserProvider', () => {
```
Comment (mldangelo) [inline test/redteam/providers/simulatedUser.test.ts]: This file name is out of date and descriptions are out of date

File: src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx [hunk 16]
```diff
@@ -542,7 +549,7 @@ export default function StrategyConfigDialog({
 
   return (
     <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
-      <DialogTitle>Configure {strategy}</DialogTitle>
+      <DialogTitle>Configure {strategyData?.name ?? strategy}</DialogTitle>
```
Comment (mldangelo) [inline src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx:624]: What does this change?

File: src/providers/promptfoo.ts (hunk 14 above)
Comment (mldangelo) [reply src/providers/promptfoo.ts:199]: Would prefer to remove it. We should stick to using the id whenever possible rather than adding new attributes.

File: src/providers/simulatedUser.ts [hunk 17]
```diff
@@ -58,40 +75,63 @@ export class SimulatedUser implements ApiProvider {
...
+      prompt: 0,
+      completion: 0,
+      numRequests: 0,
+      cached: 0,
+    };
+
+    let agentResponse: ProviderResponse | undefined;
+
     for (let i = 0; i < maxTurns; i++) {
+      logger.debug(`[SimulatedUser] Turn ${i + 1} of ${maxTurns}`);
```
Comment (mldangelo) [inline src/providers/simulatedUser.ts:128]: recommend cleaning up some of these logs

File: src/redteam/constants/metadata.ts [hunk 18]
```diff
@@ -122,6 +122,7 @@ export const subCategoryDescriptions: Record<Plugin | Strategy, string> = {
   rbac: 'Tests role-based access control implementation',
   'reasoning-dos':
     'Tests for vulnerabilities to computational resource exhaustion through excessive reasoning patterns',
+  'mischievous-user': 'Simulates a multi-turn conversation between a mischievous user and an agent',
```
Comment (mldangelo) [inline src/redteam/constants/metadata.ts]: please sort these

File: src/redteam/constants/strategies.ts [hunk 19]
```diff
@@ -11,12 +11,13 @@ export type FrameworkComplianceId = (typeof FRAMEWORK_COMPLIANCE_IDS)[number];
...
 export type DefaultStrategy = (typeof DEFAULT_STRATEGIES)[number];
 
-export const MULTI_TURN_STRATEGIES = ['crescendo', 'goat'] as const;
+export const MULTI_TURN_STRATEGIES = ['crescendo', 'goat', 'mischievous-user'] as const;
 export type MultiTurnStrategy = (typeof MULTI_TURN_STRATEGIES)[number];
 
 export const AGENTIC_STRATEGIES = [
   'crescendo',
   'goat',
+  'mischievous-user',
```
Comment (mldangelo) [inline src/redteam/constants/strategies.ts]: please keep the lists sorted

File: src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx (hunk 16 above)
Comment (will-holley) [reply src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx:624]: This fixes a bug where strategy ids are displayed instead of their names.

File: test/redteam/providers/simulatedUser.test.ts (hunk 15 above)
Comment (will-holley) [reply test/redteam/providers/simulatedUser.test.ts]: Good catch. I'm going to remove this file; it should not have been merged in.

File: src/providers/simulatedUser.ts (hunk 17 above)
Comment (will-holley) [reply src/providers/simulatedUser.ts:128]: I can do that but actually found these logs very helpful when trying to grok the state during a given turn.

File: src/providers/simulatedUser.ts (hunk 4 above)
Comment (will-holley) [reply src/providers/simulatedUser.ts]: It contextualizes the response log a few lines down, making it much easier to grok the execution state.

**Review by mldangelo — COMMENTED** [review]
This looks good

File: src/providers/simulatedUser.ts (hunk 4 above)
Comment (sklein12) [reply src/providers/simulatedUser.ts]: It doesn't work well when you're concurruency > 1

**Comment by ghost** [comment]
## ❌ Generated 10 tests - 9 passed, 1 failed (30701385b710baa80425944df8df272f939026b2) [View tests ↗](https://app.usetusk.ai/app/testing-commit-check/63885678-92f9-4c60-9fe9-45f606ecf912?client=e91e3433-8fb4-4e98-b233-bba492f4909c)

[View output in GitHub ↗](https://github.com/promptfoo/promptfoo/pull/4107/checks?check_run_id=45656308917)

### Test Summary

- `Strategies` - 4 ✅, 1 ❌
- `StrategyConfigDialog` - 5 ✅

### Results

The Tusk test suite shows mostly solid coverage for the new mischievous-user strategy implementation. 9 out of 10 tests pass, with the failing test related to warning users about incompatible strategy combinations. The tests validate critical functionality like UI rendering, configuration handling, and proper integration of the new strategy into existing components. The failing test highlights a missing feature that should be addressed before merging.

### Key Points

- ❌ `Strategies` component fails to warn users about incompatible strategy combinations (like 'mischievous-user' and 'jailbreak')
- Tests confirm proper rendering of the 'mischievous-user' strategy in the UI with correct name and description
- Configuration dialog correctly handles the new strategy's specific settings (maxTurns, stateful flag)
- Tests validate that estimated probe counts are properly capped for high test counts with the new strategy
- The implementation handles edge cases like null strategy data without crashing
- All multi-turn strategy configuration options are properly initialized with default values
<details><summary>View check history</summary>
<p>

| Commit | Status | Output | Created (UTC) |
|--------|--------|--------|--------|
| cf19fa03a10f7c17b6bf6baa3a734bd347e7bf39 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/8ffd07c3-6df4-41e2-bc6e-bbb1c6487cb3?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 6:23PM |
| 87e570e52b52307d933cf71b6e48810bd333cd81 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/98786cc5-69d2-4de8-801f-5ca4f511f32a?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 6:38PM |
| 7c7126d02d57c494da3f14f9b9e34dc1e4e563ec | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/91f24c31-899b-4ea4-b356-3a9834649730?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 6:47PM |
| 929ef9d91b8762b4764cd49ae88d36cf840ed5bf | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/40811205-7e13-4b43-a40c-d9d446c9bb93?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 7:24PM |
| 9f25aa27fc7ffa698edb1be608c6e3d65a076645 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/993cfe67-b0a7-46a2-9b07-42bea63f5b5c?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 7:30PM |
| f9f751c7429728027aa59a7ad1675d3433ce5eeb | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/9d98c11c-778d-4f30-969b-0123be5f7ad6?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 7:31PM |
| 8a6e6f2f4a9ce6649247a0ea8ee45a7d401fcf00 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/2334d66b-476c-4e52-9ee6-2bced74ef2a3?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 7:33PM |
| e98d020e06b226e110cde769b15cc7f9c942bab1 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/c400408b-8a4f-4a14-95b8-a0b60c206dcf?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 7:59PM |
| 02d48611b3ac6d2d5b13cdd4125eebee0ad5c4fb | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/7964b2df-0923-460d-ab7a-5d88769cfc4a?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 8:14PM |
| 5b5e1a45f30c4e47731ea97bcf622ed14e7b592c | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/01d925ff-363b-4a8d-bc30-fd99c8ca6718?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 20, 2025 8:33PM |
| 099e7c8ae0ed728f792538616d2bccffd77fa857 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/795d833f-913c-4acc-86ed-8216b871415b?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 21, 2025 5:38PM |
| 985592278c8f24eb0efc7688b67e09f264505023 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/378b9efe-6cd6-4efb-b3da-30bd41bc5693?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 21, 2025 5:43PM |
| 07fe36003067860bf5fb25b6ad1bcb9ab613e4a8 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/58f61dae-0ed3-438c-9955-5d14d1424c0c?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 23, 2025 12:48PM |
| 88f8b457a3f738337c3384c3bc74b450d73f66b7 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/1f9a9a51-2921-453a-a0e0-05081fb33395?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 23, 2025 5:08PM |
| e9248b22b249b119a41c9338bc06ede1f407b23b | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/617ea140-e19a-4d3b-a1b2-7e06739678e2?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 3:22PM |
| 343f38edd3349c9bc2600478d86f64f687bd6bcf | ✅ Generated 6 tests - 6 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/ec7e503b-c485-4eae-8bd3-b5ceb85ca36b?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 7:13PM |
| 62260ad6a2436a0f8a966301613256815e4257ce | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/fffa5a5d-4d3f-48c3-8be6-7b570db945b3?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 9:37PM |
| 2204f09cc86679bdca80a095b11a57e7be641c12 | ⏩ Skipped due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/83472be4-8375-4a4a-8f10-ee7201bb73d1?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 9:59PM |
| 3aa90e5ee6494462478f98440472a2cf12be76d0 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/72babe95-c93b-4691-b1f2-92ad61c088d3?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 10:02PM |
| 8f703311d7e84f9fccb56fc3e9fdc2c258769470 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/c8b0cb90-fb2b-4215-a026-89291e4a125a?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 25, 2025 10:11PM |
| 3c58cfdc20fc83d8db09139bfc327ddaea07b77a | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/13d4dc83-3139-42fd-94b6-8683f1483eed?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 26, 2025 7:22PM |
| 4bb061771761052031d8dc8e0bb5cd1ba1721f4b | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/5f206ec6-ab5b-4823-92a1-8dc077c050e8?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 26, 2025 8:09PM |
| 480a44a77e46d67548cde9b1590e733730500d61 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/e7daee2e-bfe1-4cb1-b66d-f42ac187a4ea?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 27, 2025 2:05PM |
| 11950ea66ea27a5990d3a26f782a1f51f6e7ed4f | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/a78067cc-5c81-4f2a-841a-78d4ee2d2346?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 27, 2025 6:47PM |
| 7869ed7bd22ea17bb357962fc99cf26f16f3ec17 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/96e407aa-a039-421e-8e9f-8b251132b4cb?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 27, 2025 6:56PM |
| 7f7d7f37fb31cf3a4e0d523ad811706b7d260fff | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/f4413ec7-ad0b-4c2e-a030-b79850e0a161?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 30, 2025 7:01PM |
| 0cad0423e537e32fa39230d709f5dac57cca28d6 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/a2c9abbc-f080-4607-bc84-082a10725bed?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 30, 2025 7:12PM |
| ebcebc350dee6092c0e8165b39384b07e3c02d3a | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/6e222174-e7c8-4224-bd2a-f6a474c04d00?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jun 30, 2025 7:15PM |
| c35ddc5dee4b61705a4548d9a2b0af8e5a0be5ac | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/f711ae4e-2ece-4770-88c1-57f8c30ba42d?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 1, 2025 9:26PM |
| f4d916bcc4e480a4d6de25e95838fd7b3ed0ac6d | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/35719353-9ba9-41c1-8956-69bc4a75abeb?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 1, 2025 9:44PM |
| ede381b215826bd05c47c7082ebad12e50942bbf | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/4b4cabfd-54ea-4e71-a39f-43406c82de54?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 2, 2025 2:33PM |
| 3f4004c398f978852a1e7d812a6b24eec19eba83 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/2de15fee-35d6-4f57-b4e4-91987d5474d6?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 2, 2025 3:23PM |
| 4cbdb24a31f7d83d1d5ccf81621017f28259eb97 | ✅ Generated 2 tests - 2 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/5dc97450-b383-4c35-93e3-e1c6efcfb228?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 3, 2025 3:26PM |
| 9dd8b044c1a5200f16a5a645ae9356615c1fadc8 | ⏩ Cancelled due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/9dc4faec-2f3d-4eeb-a84a-400e9f7818fc?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 3, 2025 7:59PM |
| ec950a0425b9d6b1f2fe79ecfe3e54b081fff8c8 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/76d1a8ad-daec-4740-839a-87cdd843be80?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 3, 2025 8:04PM |
| 2f37f64a08e819e23ab590d4672f22758ee69c97 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/81c896b7-ad04-4697-92c6-21b0c69afe45?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 3, 2025 8:09PM |
| 36f8b5ca3259aa633d9be7d2804036d314c0a4e3 | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/36237d54-b220-46ab-bad3-5f7d5406df08?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 3, 2025 8:24PM |
| d15200ab0c2795de78bf0a9a4e593d381facbff1 | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/c7c089a4-0cc3-4909-9edd-d69ba8aed75b?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 4:35PM |
| 0f36362e7f8de8678bcb78e3d3df0aaa8ac6345b | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/d70fc0b6-712c-49bb-81e5-3406f40f882d?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 5:07PM |
| c82314e8680c2e20519dd38dbd1862d9e5c28ae0 | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/f8248967-4a1f-421e-9d75-4d75e0e91e3f?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 7:37PM |
| 6289e8fcef0aa2e362fca9366e2951ec40bba67f | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/543170b4-1d7d-4de2-96c4-c508e405f3fe?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 8:25PM |
| aa4003df13fbf26cef2351881935cfdbd44c296e | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/7480efba-230b-4428-a486-05bb717a8c65?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 8:26PM |
| 124c78eb4df77c6b2c129a1a62fcba63f00f442d | ❌ Generated 9 tests - 8 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/5f77d269-2825-4a16-8f27-a97c09bbee2e?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 8:27PM |
| de622f20e9b42fed40b5e56df9ad13d5274dc8b6 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/996cd6c6-cf3d-4b85-8022-63f959348a09?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 8:34PM |
| ae5fb5d5904b11433fd58de55dd72afaa9eb54c1 | ❌ Generated 7 tests - 6 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/9835782a-2100-4e49-b12e-5851b1904544?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 8:48PM |
| bb0e48c6446890e737934df715adece77522ba57 | ⏩ Skipped due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/06cf8d3a-24ec-4df7-9f11-db0d24145df9?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 10:07PM |
| df755510e8b5fdf7672aa6523c825df6cffc142d | ❌ Generated 7 tests - 6 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/ca491ae2-ab00-4a2a-b146-8a5b752841b8?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 10:08PM |
| b07121975c7bda5a96fe6ef94539c33dda6e59a2 | ❌ Generated 7 tests - 6 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/fc41475d-09f5-410c-a1db-a6b7efda24bc?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 7, 2025 10:19PM |
| f30fc793970eee1971f7c983fca83739d75511be | ❌ Generated 10 tests - 9 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/3cdf304b-0312-443b-9916-26f1fc586d1f?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 8, 2025 12:46PM |
| 26640e38b878525376f60b5da599f0582579da75 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/185ed8f0-fbb4-49b0-a901-de87bbcaef8e?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 8, 2025 1:26PM |
| 6c0d62a0060bb49911557cd6d8c306446e63e5f7 | ❌ Generated 10 tests - 9 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/0f8c44d6-d365-4a91-be75-a4814e4301e9?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 8, 2025 1:32PM |
| 39216bb738aa9b92b09e90976cd92edc2f5e6d8d | ❌ Generated 10 tests - 9 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/293b0f0a-2109-46d4-9a1f-dd81ca44dedb?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 9, 2025 3:43PM |
| 30701385b710baa80425944df8df272f939026b2 | ❌ Generated 10 tests - 9 passed, 1 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/63885678-92f9-4c60-9fe9-45f606ecf912?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jul 9, 2025 4:10PM |

</p>
</details>

_Was Tusk helpful? Give feedback by reacting with 👍 or 👎_
